remotes/handlers.go

```diff
@@ -206,3 +206,38 @@ func PushContent(ctx context.Context, pusher Pusher, desc ocispec.Descriptor, pr
 
 	return nil
 }
+
+// FilterManifestByPlatformHandler allows Handler to handle non-target
+// platform's manifest and configuration data.
+func FilterManifestByPlatformHandler(f images.HandlerFunc, m platforms.Matcher) images.HandlerFunc {
+	return func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+		children, err := f(ctx, desc)
+		if err != nil {
+			return nil, err
+		}
+
+		// no platform information
+		if desc.Platform == nil || m == nil {
```

Comment:
I had some concerns about this but not sure if I should dismiss them. My concern is that a manifest which just doesn't set any platform will end up having the default behavior to pull all objects. My first solution was to only do this for the first manifest seen with no platform.

Maybe we need another function that can filter out "non-image" manifests by either limiting the number of unspecified platforms pulled and ensuring the configuration types are understood.